src/control/server/mgmt_client.go

```diff
@@ -59,6 +59,26 @@ func newMgmtSvcClient(ctx context.Context, log logging.Logger, cfg mgmtSvcClient
 	}
 }
 
+// delayRetry delays next retry.
+func (msc *mgmtSvcClient) delayRetry(ctx context.Context) {
```

Comment:
This use of context.WithTimeout is needlessly complicated IMO. I propose just using time.After(retryDelay), e.g.
```go
func (msc *mgmtSvcClient) delayRetry(ctx context.Context) {
        select {
        case <-ctx.Done():  // break early if the parent context is canceled
        case <-time.After(retryDelay): // otherwise, block until after the delay duration
        }
}
```